src/InstrumentationEngine.Lib/InstrumentationMethod.cpp

```diff
@@ -83,8 +83,20 @@ HRESULT MicrosoftInstrumentationEngine::CInstrumentationMethod::InitializeCore(
     WCHAR wszModuleFullPath[MAX_PATH];
     memset(wszModuleFullPath, 0, MAX_PATH);
     wcscpy_s(wszModuleFullPath, MAX_PATH, m_bstrModuleFolder);
-    PathCchAppend(wszModuleFullPath, MAX_PATH, m_bstrModule);
 
+    // For Windows 7 support, we cannot use PathCch functions.
+    //PathCchAppend(wszModuleFullPath, MAX_PATH, m_bstrModule);
+    WCHAR wszBuffer[2 * MAX_PATH]; // set to 2x the MAX_PATH in order to prevent buffer overflow.
+    if (!PathCanonicalize(wszBuffer, wszModuleFullPath) ||
```

Comment:
>wszModuleFullPath [](http://example.com/codeflow?start=37&length=17)

can you just use m_bstrModuleFolder and avoid the string copy into wszModuleFullPath above?